app/tandem/SEAS.cpp

```diff
@@ -135,16 +135,51 @@ template <typename T> struct qd_operator_specifics {
 
     static void print_profile(T const&) {}
 };
+
 template <>
-struct operator_specifics<SeasQDOperator> : public qd_operator_specifics<SeasQDOperator> {};
+struct operator_specifics<SeasQDOperator>
+  : public qd_operator_specifics<SeasQDOperator> {};
+
+
+template <typename T> struct qd_green_operator_specifics {
+  using monitor_t = seas::MonitorQD;
+
+  static auto make(LocalSimplexMesh<DomainDimension> const& mesh, Config const& cfg, seas::ContextBase& ctx) {
+    std::string prefix;
+    if (cfg.gf_checkpoint_prefix) { prefix = *(cfg.gf_checkpoint_prefix); }
+    else { prefix = ""; }
+    auto seasop = std::make_shared<T>(std::move(ctx.dg()), std::move(ctx.adapter()),
+                                      std::move(ctx.friction()), mesh, cfg.matrix_free,
+                                      MGConfig(cfg.mg_coarse_level, cfg.mg_strategy), prefix);
+    ctx.setup_seasop(*seasop);
+    seasop->warmup();
+    return seasop;
+  }
```

Comment:
So you only change the make function. You could specialize operator_specifcs (as done for SeasFDOperator) and inherit from qd_operator_specifics such that you don't need to duplicate code.

Reply:
done